btree/btree.c

```diff
@@ -2939,145 +2953,128 @@ static void ff_capture(struct slot *slot, struct m0_be_tx *tx)
 	M0_BTREE_TX_CAPTURE(tx, seg, h, hsize);
 }
 
-static void ff_node_free_credits(const struct nd *node,
-				 struct m0_be_tx_credit *accum)
+/**
+ * This function will calculate credits required to allocate the node and it
+ * will add those credits to @cred.
+ */
+static void ff_node_alloc_credit(const struct nd *node,
+				 struct m0_be_tx_credit *cred)
 {
-	struct ff_head *h           = ff_data(node);
-	int             shift       = h->ff_shift;
-	int             node_size   = 1ULL << shift;
-	int             header_size = sizeof(*h);
-
-	m0_be_allocator_credit(NULL, M0_BAO_FREE_ALIGNED,
-			       node_size, shift, accum);
+	int         shift     = ff_shift(node);
+	m0_bcount_t node_size = 1ULL << shift;
 
-	m0_be_tx_credit_add(accum, &M0_BE_TX_CREDIT(1, header_size));
+	m0_be_allocator_credit(NULL, M0_BAO_ALLOC_ALIGNED,
+			       node_size, shift, cred);
 }
-#ifndef __KERNEL__
+
 /**
- *  --------------------------------------------
- *  Section START - Btree Credit
- *  --------------------------------------------
+ * This function will calculate credits required to update the node and it will
+ * add those credits to @cred.
  */
-/**
- * Credit for btree put and delete operation.
- * For put operation, at max 2 nodes can get captured in each level plus an
- * extra node. The delete operation can use less nodes, still use the same api
- * to be on the safer side.
- *
- * @param accum transaction credit.
- */
-static void btree_callback_credit(struct m0_be_tx_credit *accum)
+static void ff_node_update_credit(const struct nd *node,
+				   struct m0_be_tx_credit *cred)
 {
-	accum->tc_cb_nr += BTREE_CALLBACK_CREDIT;
+	int         shift     = ff_shift(node);
+	m0_bcount_t node_size = 1ULL << shift;
+
+	m0_be_tx_credit_add(cred, &M0_BE_TX_CREDIT(1, node_size));
```

Comment:
s/1/2/
If you look at ff_capture then it calls the CAPTURE routine twice based on a condition. So we can keep the same node_size but the region count should be 2.
Also make similar changes for ff_put_credit() and ff_del_credit() if needed.

Reply:
done.